### PR TITLE
Devel logical expr

### PR DESCRIPTION
**Pull request by saidctb**
In this PR we made sure that Psydac works with the new version of Sympde and Pyccel where :
- we introduced the concept of a mapped domain and a logical domain
- the expression passed to the parser are defined in the logical domain only
- update the api tests
- support sympy 1.5 and 1.6

### CODE REVIEW

File: psydac/api/tests/test_api_glt_2d_scalar.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from sympde.core import Constant
 from sympde.calculus import grad, dot, inner, cross, rot, curl, div
 from sympde.calculus import laplace, hessian
-from sympde.topology import (dx, dy, dz)
+from sympde.topology import dx, dy, dz, dx1, dx2, dx3
```
Comment (yguclu) [inline psydac/api/tests/test_api_glt_2d_scalar.py:9]: Please remove the import of `(dx, dy, dz)`, as they are never used in this file.

**Review by yguclu — CHANGES_REQUESTED** [review]
Great job @saidctb! It only remains to remove `mapping` from this line,
```python
Vh = discretize(V, domain_h, mapping=mapping, degree=degree)
```
because the mapping is now attached to `domain_h`.

File: psydac/api/tests/test_api_glt_2d_scalar.py (hunk 1 above)
Comment (saidctb) [reply psydac/api/tests/test_api_glt_2d_scalar.py]: done

**Review by yguclu — DISMISSED** [review]

File: psydac/api/ast/parser.py [hunk 2]
```diff
@@ -5,22 +5,24 @@
 from sympy import Add
 from sympy import Abs
 from sympy import Symbol, Idx
+from sympy import Max
+from sympy import Basic
+from sympy.simplify import cse_main
```
Comment (yguclu) [inline psydac/api/ast/parser.py:10]: It appears that this function is not currently used in the module. Do you still want to import it?

Comment (saidctb) [reply psydac/api/ast/parser.py:10]: ah I forgot to push my commits

**Review by yguclu — APPROVED** [review]

**Pull request merged**